src/docs/api/design.mdx

```diff
@@ -24,3 +25,54 @@ In the above example we flattened `projects`. The table below shows the existing
 | projects | When the resource is under a specific project in the hierarchy but not under an issue | organizations | {organization_slug}/ {project_slug} | [Create a New Client Key](https://docs.sentry.io/api/projects/create-a-new-client-key/)|
 | issues | When the resource is under a specific issue in the hierarchy | projects | {issue_id} | [List an Issue's Events](https://docs.sentry.io/api/events/list-an-issues-events/)|
 | sentry-app-installations | When the resource is mapped to a specific integration | organizations | {integration_slug} | [Delete an External Issue](https://docs.sentry.io/api/integration/delete-an-external-issue/)|
+
+Here are some additional examples:
+- 👍 `/organizations/{organization_slug}/projects/`
+- 👍 `/projects/{organization_slug}/{project_slug}/issues/`
+- 👎 `/projects/`
+
+**Note:** Hierarchy here does not necessarily mean that one collection belongs to a parent collection. For example `projects/{project_identifier}/teams/` and `teams/{team_identifier}/projects/` can co-exist. The first one means the teams that are working on a specific project and the latter means the projects one specific team is working on.
+
+### Naming Guidance
+- Collection name should be hyphen-ated lower case, e.g. commit-files.
+- Collection names must be plural. Avoid using uncountable words because user can’t know if the GET returns one item or a list.
+- Query params and body params are CamelCased. 
+- For sorting and filtering try to stick with the common param names: sortBy (e.g. date_created), orderBy (either asc or desc) groupBy, limit 
+- Path params are snake_cased
+
+## Functionality
+Each API should be **stateless**, have a clear purpose and do one clear thing. To achieve that try to stick with the standard methods below. If you have a need to a more complicated functionality reach out to [owners-api](https://github.com/orgs/getsentry/teams/owners-api) and we can help you achieve it.
+
+- 👍 An API that updates project settings: PATCH or PUT
```

Comment:
Do we want to distinguish between the two? I.e. `PATCH` is replacing select field, `PUT` replaces the whole object